### ISSUE TITLE
How to label implementation tasks

### ISSUE THREAD

**Issue by lcabrini** [lcabrini/aluguntugui#10]
I cannot find a suitable label for implementation of a task related to a spec. Do we add one, or use an existing one?

**Comment by benbaka** [comment]
We will create a label called "In development". What do you think  ?

**Comment by lcabrini** [comment]
That might be it. So idea -> spec -> in development -> ?

**Comment by lcabrini** [comment]
How about we call it "implementation" instead? It feels a bit more consistent with "idea" and "spec". Or what do you think?

**Comment by benbaka** [comment]
Implementation does best describe the phrase "in development".

**Comment by benbaka** [comment]
By the way, why are you removing the labels and placing new ones on the issues ? If you leave them there, it gives better visual of the stages the issue has been through. Else, we will have to define a different document to describe the flow for new people joining the development team and they will have to imbibe it.
Also, if a label now needs to move back one stage, slapping another label on it will tell everyone the evolution of the issue, visually. 
Thoughts ?

**Comment by benbaka** [comment]
sorry but I think I now understand the flow you have adopted.

**Comment by lcabrini** [comment]
Good point! Will do that.

**Comment by benbaka** [comment]
I am still find it hard to make sense of the flow. I need something to help me know when an issue is metamorphosing into another thing. Please slap labels on issues when they get to various stages. 
I am struggling with keeping something like the "create login" issue when it was in idea stage and spec stage and now implementation stage. I want to have one coherent page that lets me know all that has happened with the "create login" issue. I don't like the clicking around too much. 

Thoughts ?

**Comment by lcabrini** [comment]
But i did use idea and spec labels on the login issue. And in fact, that issue is still at spec stage, because we have not finalized and agreed on the spec. Maybe I should not have added the implementation tasks yet, but I was also curious to see how the workflow is going to evolve. 

In a way, I think it would be better to use this workflow:

1. an issue is raised for an idea. We discuss that idea and decide whether or not it is a good idea.
2. once we feel that the idea is viabla a NEW issue is raised to write the corresponding spec. Once this is done, we could close the idea issue. The spec issue will have less discussion because we should have exhausted most of that at idea stage, but other questions may come up.
3. Once the spec is fully agreed upon, we create the implementation tasks that work towards that particular spec. These should link to the spec (as you noted).
4. Once all the implementation tasks towards a spec are completed that spec is considered implemented. Now, any change relating to that spec would require a new version of the spec itself

This would mean more issues, since we would have different issues for idea and spec. But I think it would make the workflow clearer. What do you think?

**Comment by benbaka** [comment]
Ok. I think that is ok. I will have to warm up to this workflow. 
It is beginning to look like standardization efforts are even difficult than programming. I will advice that we do not rush this process because it is very important. We are laying the foundation and us such, requires much time. And it is not also easy. 
I am not deterred about our slowness. It rather gives me confidence that everything that we are doing is being carefully thought out.

**Comment by lcabrini** [comment]
Yes, those processes are just as important as the programming/implementation. Successful teams are partly successful because they get the workflow right. Unsuccessful teams focus too much time on rewriting parts of a system that has not been well thought out.

I agree with you. Let's make this our own learning experience. Whatever mistakes we make along the way will increase our wisdom and experience.
